stbt-run

```diff
@@ -22,7 +25,7 @@ parser.add_argument(
     '--save-video', help='Record video to the specified file', metavar='FILE',
     default=stbt.get_config('run', 'save_video'))
 parser.add_argument(
-    '--save-trace', metavar='FILE', default=None,
+    '--save-trace', metavar='FILE', nargs='?', const=TO_SOCKET,
```

Comment:
I don't think we need this 'TO_SOCKET' sentinel.  `trace_test_execution` can decide whether to install the trace handler based on whether the `STBT_TRACING_SOCKET` environment variable is set.  In general I'm not a fan of using values that would otherwise be valid as sentinels.  You may think that it's unlikely that someone would want to log to a file called `TO_SOCKET` but the presence of this special value can be the cause of confusing failures, particularly when `stbt-run` is being scripted.